internal/app/referenceclient/impl.go

```diff
@@ -341,32 +371,51 @@ func (i *invoker) clientStream(
 		trailers = internal.ConvertToProtoHeader(resp.Trailer())
 	}
 
+	wireDetails, err := getWireDetails(ctx)
+	if err != nil {
+		return nil, err
+	}
+
 	return &v1.ClientResponseResult{
-		ResponseHeaders:  headers,
-		ResponseTrailers: trailers,
-		Payloads:         payloads,
-		Error:            protoErr,
-		ConnectErrorRaw:  nil, // TODO
+		ResponseHeaders:    headers,
+		ResponseTrailers:   trailers,
+		Payloads:           payloads,
+		Error:              protoErr,
+		ActualStatusCode:   wireDetails.StatusCode,
+		ActualHttpTrailers: wireDetails.Trailers,
+		ConnectErrorRaw:    wireDetails.ConnectErrorRaw,
 	}, nil
 }
 
 func (i *invoker) bidiStream(
 	ctx context.Context,
 	req *v1.ClientCompatRequest,
-) (result *v1.ClientResponseResult, _ error) {
+) (result *v1.ClientResponseResult, err error) {
 	ctx, cancel := context.WithCancel(ctx)
 	defer cancel()
 
 	result = &v1.ClientResponseResult{
 		ConnectErrorRaw: nil, // TODO
 	}
 
+	ctx = withWireCapture(ctx)
+
 	stream := i.client.BidiStream(ctx)
 	defer func() {
+		wireDetails, e := getWireDetails(ctx)
+		if e != nil {
+			result = nil
+			err = e
+			return
+		}
```

Comment:
I want to call `getWireDetails` in the `defer` function because it's much easier to reason about in the bidi code. But, wasn't sure if this was the best way to handle the error.

Reply:
A few things:
1. I think you only want to call `getWireDetails` when `err == nil`. If this function is already returning an error (and not a result), `getWireDetails` is about guaranteed to fail, and this would also overwrite the existing error that is being returned.
2. When you get an error, I think you'd want to instead store it in [`ClientCompatResponse.feedback`](https://buf.build/connectrpc/conformance/docs/main:connectrpc.conformance.v1#connectrpc.conformance.v1.ClientCompatResponse). Admittedly, that's a bit of a pain to do with the current structure. Maybe you need something like a `internal.Printer` to collect feedback messages and pass that into `invoke` (and then into this method). Then this function could write the error message into that printer. Then the caller, when it's creating the `ClientCompatResponse` (using either the result or error that this returns), can add those messages to that field. _OR_ we could just move the `feedback` field _into_ the `ClientResponseResult` message. WDYT? (I added it as a "just in case", to have feedback from the reference client like we have feedback from the reference server; but it's not yet used.)

Reply:
Though... I'm slightly worried that maybe an error getting those details isn't something that should fail a conformance test case. Like maybe it indicates a bug in the reference client _not_ in the server under test. So in that case, you might just eat it and put a TODO for what to do with it.